llama_index/utils.py

```diff
@@ -239,12 +248,16 @@ def get_transformer_tokenizer_fn(model_name: str) -> Callable[[str], List[str]]:
     return tokenizer.tokenize
 
 
-def get_cache_dir() -> Path:
+def get_cache_dir() -> str:
     """Locate a platform-appropriate cache directory for llama_index,
     and create it if it doesn't yet exist
     """
+    # User override
+    if "LLAMA_INDEX_CACHE_DIR" in os.environ:
+        path = Path(os.environ["LLAMA_INDEX_CACHE_DIR"])
+
     # Linux, Unix, AIX, etc.
-    if os.name == "posix" and sys.platform != "darwin":
+    elif os.name == "posix" and sys.platform != "darwin":
         # use ~/.cache if empty OR not set
         base = os.path.expanduser("~/.cache")
```

Comment:
Instead of setting the default to `/home` or `~/`, I think it better to set it to `/tmp`

https://github.com/jerryjliu/llama_index/pull/6579